enumerable.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+module Enumerable
+  def my_each()
+    return self unless block_given?
+
+    length.times do |index|
+      yield(self[index])
+    end
+  end
+
+  def my_each_with_index()
+    return self unless block_given?
+
+    length.times do |index|
+      yield(self[index], index)
+    end
+  end
+
```

Comment:
- [ ] This should also return an enumerator if no block is given.

- [ ] It also should return self if the block is given and self is an array

- [ ] when a block is given and  when self is a range, it should return the range itself after calling the given block once for each element in self

- [ ]  when a block is given, and when self is a range, it behaves like #each_with_index calls the given block once for each element in self